apps/console/src/features/organizations/components/edit-organization-role/edit-organization-role-basic.tsx

```diff
@@ -0,0 +1,434 @@
+        //     regEx = PRIMARY_USERSTORE_PROPERTY_VALUES.RolenameJavaScriptRegEx;
+        // }
+        return PRIMARY_USERSTORE_PROPERTY_VALUES.RolenameJavaScriptRegEx;
+    };
+
+    /**
+     * The following function validates role name against the user store regEx.
+     */
+    const validateRoleNamePattern = async (): Promise<string> => {
+        let userStoreRegEx = "";
+
+        if (userStore !== SharedUserStoreConstants.PRIMARY_USER_STORE) {
+            await SharedUserStoreUtils.getUserStoreRegEx(userStore,
+                SharedUserStoreConstants.USERSTORE_REGEX_PROPERTIES.RolenameRegEx)
+                .then((response) => {
+                    setRegExLoading(true);
+                    userStoreRegEx = response;
+                });
+        } else {
+            await SharedUserStoreUtils.getPrimaryUserStore().then((response) => {
+                setRegExLoading(true);
+                if (response && response.properties) {
+                    userStoreRegEx = response?.properties?.filter(property => {
+                        return property.name === "RolenameJavaScriptRegEx";
+                    })[ 0 ].value;
+                }
+            });
+        }
+
+        setRegExLoading(false);
+
+        return new Promise((resolve, reject) => {
+            if (userStoreRegEx !== "") {
+                resolve(userStoreRegEx);
+            } else {
+                reject("");
+            }
+        });
+
+    };
+
+    /**
+     * The following function handles the role name change.
+     *
+     * @param event
+     * @param data
+     */
+    const handleRoleNameChange = (event: ChangeEvent, data: InputOnChangeData): void => {
+        setIsRoleNamePatternValid(SharedUserStoreUtils.validateInputAgainstRegEx(data?.value, userStoreRegEx));
+    };
+
+    /**
+     * Dispatches the alert object to the redux store.
+     *
+     * @param {AlertInterface} alert - Alert object.
+     */
+    const handleAlerts = (alert: AlertInterface): void => {
+        dispatch(addAlert(alert));
+    };
+
+    /**
+     * Function which will handle role deletion action.
+     *
+     * @param id - Role ID which needs to be deleted
+     */
+    const handleOnDelete = (id: string): void => {
+        deleteOrganizationRole(currentOrganization.id, id).then(() => {
+            handleAlerts({
+                description: t("console:manage.features.roles.notifications.deleteRole.success.description"),
+                level: AlertLevels.SUCCESS,
+                message: t("console:manage.features.roles.notifications.deleteRole.success.message")
+            });
+            if (isGroup) {
+                history.push(AppConstants.getPaths().get("GROUPS"));
+            } else {
+                history.push(
+                    AppConstants.getPaths().get("ORGANIZATION_ROLES")
+                );
+            }
+        });
+    };
+
+    /**
+     * Method to update role name for the selected role.
+     *
```

Comment:
Remove unnecessary line

Reply:
Fixed in https://github.com/wso2/identity-apps/pull/3617/commits/81488c55dd6c9c7a63575dee78e71f2b80643963